tgtg_scanner/notifiers/smtp.py

```diff
@@ -79,29 +81,61 @@ def _stay_connected(self) -> None:
         if status != 250:
             self._connect()
 
-    def _send_mail(self, subject: str, html: str) -> None:
+    def _send_mail(self, subject: str, html: str, item_id: int) -> None:
         """Sends mail with html body"""
         if self.server is None:
             self._connect()
         if self.sender is None or self.recipients is None or self.server is None:
             raise SMTPConfigurationError()
         message = MIMEMultipart("alternative")
         message["From"] = self.sender
-        message["To"] = ", ".join(self.recipients)
+
+        # Contains either the main recipient or recipient(s) that should be
+        # notified for the specific item
+        recipients = []
+        item = str(item_id)
+
+        # Determine recipient(s) based on item_id
+        if self.recipients_per_item:
+            try:
```

Comment:
At this point, the configuration should be valid and no longer need to be checked. It is also unnecessary to check the configuration every time before sending, as it cannot change in the meantime. Therefore, please implement an appropriate check somewhere below line 37. Simultaneously, the configuration should be transferred into a variable with appropriate typing.

suggestion (untested):
````python
self.item_recipients: Union[dict[str, list[str]], None] = None
if config.smtp.recipients_per_item is not None:
  json_data = json.loads(config.smtp.recipients_per_item)
    if not isinstance(json_data, dict) or any(not isinstance(value, list) for value in json_data.values):
      raise SMTPConfigurationError("Recipients per Item is not a valid dictionary")
   self.item_recipients = json_data
````

Reply:
Thank's for Yor feedback! I moved the syntax validation alongside the check for list and string format of mail addresses to the `init()` method below the check for `self.enabled`. The functionality of the method for sending mail only determines the recipient(s) depending on the given item, from now on.